types/cometd/index.d.ts

```diff
@@ -81,12 +82,57 @@ export interface Configuration {
     useWorkerScheduler?: boolean;
 }
 
-export interface Message {
+export type ConnectionType = 'long-polling' | 'callback-polling' | 'iframe' | 'flash';
+export type ReconnectAdvice = 'retry' | 'handshake' | 'none';
+
+export interface BaseMessage {
     successful: boolean;
-    data: any;
+    channel: string;
+    id?: string;
```

Comment:
This is a lot of optional fields. None of these are guaranteed?

Reply:
correct